src/cartridge/src/meta_cartridge.cpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file meta_cartridge.cpp
+ * Defines the MetaCartridge class
+ */
+
+#include "cartridge/meta_cartridge.h"
+
+#include <string>
+#include <vector>
+
+#include <iomanip>
+#include <iostream>
+
+namespace cartridge {
+
+MetaCartridge::MetaCartridge() {}
+
+void MetaCartridge::populate_meta(std::vector<uint8_t> &data) {
+	title = std::string(data.begin() + 0x0134, data.begin() + 0x0143);
+	cgb_flag = data[0x0143];
+	old_licensee_code = data[0x014B];
+	sgb_flag = data[0x0146];
+	cartridge_type = data[0x0147];
+	rom_size = data[0x0148];
+	cartridge_ram = data[0x0149];
+	dest_code = data[0x014A];
+}
+
+void MetaCartridge::check_logo_validity(std::vector<uint8_t> &data) {
+	Address start_address = 0x0104;
+	for (auto i = 0; i < 48; i++) {
+		if (data[start_address + i] != nintendo_logo[i]) {
+			is_logo_valid = false;
+			break;
+		}
+	}
+}
+
+void MetaCartridge::display_meta() {
```

Comment:
Can you move this method into cartridge, and call it from cartridge, passing the metadata object? Seems cleaner than doing a cout from inside our struct here.